Sensors/SMCDellSensors/SMIMonitor.cpp

```diff
@@ -9,21 +9,31 @@
  */
 
 #include "SMIMonitor.hpp"
+#include "kern_hooks.hpp"
+
 #include <Headers/kern_cpu.hpp>
 
 extern "C" {
 #include <i386/pmCPU.h>
 }
 
 SMIMonitor *SMIMonitor::instance = nullptr;
+_Atomic(bool) volatile SMIMonitor::smmIsBeingRead = false;
 
 OSDefineMetaClassAndStructors(SMIMonitor, OSObject)
 
 int SMIMonitor::i8k_smm(SMMRegisters *regs) {
 	int rc;
 	int eax = regs->eax;  //input value
 	
-	IOSimpleLockLock(preemptionLock);
+	smmIsBeingRead = true;
```

Comment:
Use atomic_load/store_explicit here and everywhere else and drop volatile? If you do not want to bother about memory ordering could just use the sequential consistency for now.